File: attendy-v2/frontend/src/types/attendance.ts
import type { ClassSection } from './index'

export interface AttendanceRow {
  student_id: string
  name: string
  roll_number: number
  class_section: ClassSection
  status: 'present' | 'absent'
  event_time: string | null
  confidence: number | null
  source: string | null
}

export interface AttendanceSheetResponse {
  date: string
  items: AttendanceRow[]
  total: number
  present_count: number
  absent_count: number
}

export interface AttendanceConfirmedEvent {
  type: 'attendance_confirmed'
  student_id: string
  name: string
  roll_number: number
  class_section: ClassSection
  event_time: string
  confidence: number
  source: string
}

File: attendy-v2/backend/app/schemas/attendance.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel

from app.schemas.class_section import ClassSectionOut


class AttendanceRow(BaseModel):
    student_id: uuid.UUID
    name: str
    roll_number: int
    class_section: ClassSectionOut
    status: str  # "present" | "absent"
    event_time: datetime | None
    confidence: float | None
    source: str | None


class AttendanceSheetResponse(BaseModel):
    date: date
    items: list[AttendanceRow]
    total: int
    present_count: int
    absent_count: int


class ManualAttendanceRequest(BaseModel):
    student_id: uuid.UUID
    event_date: date
    status: str  # "present" | "absent"


class DailySummaryPoint(BaseModel):
    date: date
    present_count: int
    absent_count: int
    total: int


class AttendanceSummaryResponse(BaseModel):
    date_from: date
    date_to: date
    points: list[DailySummaryPoint]
    overall_present_rate: float


class ChronicAbsenteeRow(BaseModel):
    student_id: uuid.UUID
    name: str
    roll_number: int
    class_section: ClassSectionOut
    school_days: int
    absent_days: int
    absence_rate: float


class ChronicAbsenteesResponse(BaseModel):
    date_from: date
    date_to: date
    threshold: float
    items: list[ChronicAbsenteeRow]
